com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderUtil.java

```diff
@@ -145,6 +147,124 @@ public static ArrayList<Resource> getInputs() {
+                new ShaderDataTypeConversionEntry("float", ShaderDesc.ShaderDataType.SHADER_TYPE_FLOAT),
+                new ShaderDataTypeConversionEntry("vec2", ShaderDesc.ShaderDataType.SHADER_TYPE_VEC2),
+                new ShaderDataTypeConversionEntry("vec3", ShaderDesc.ShaderDataType.SHADER_TYPE_VEC3),
+                new ShaderDataTypeConversionEntry("vec4", ShaderDesc.ShaderDataType.SHADER_TYPE_VEC4),
+                new ShaderDataTypeConversionEntry("mat2", ShaderDesc.ShaderDataType.SHADER_TYPE_MAT2),
+                new ShaderDataTypeConversionEntry("mat3", ShaderDesc.ShaderDataType.SHADER_TYPE_MAT3),
+                new ShaderDataTypeConversionEntry("mat4", ShaderDesc.ShaderDataType.SHADER_TYPE_MAT4),
+                new ShaderDataTypeConversionEntry("sampler2D", ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER2D),
+                new ShaderDataTypeConversionEntry("sampler3D", ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER3D),
+                new ShaderDataTypeConversionEntry("samplerCube", ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER_CUBE),
+                new ShaderDataTypeConversionEntry("sampler2DArray", ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER2D_ARRAY)));
+
+        public static ShaderDesc.ShaderDataType stringTypeToShaderType(String typeAsString) {
+
+            for (ShaderDataTypeConversionEntry e : shaderDataTypeConversionLut) {
+                if (e.asStr.equals(typeAsString)) {
+                    return e.asDataType;
+                }
+            }
+            return ShaderDesc.ShaderDataType.SHADER_TYPE_UNKNOWN;
+        }
+
+        public static String shaderTypeToString(ShaderDesc.ShaderDataType dataType) {
+            for (ShaderDataTypeConversionEntry e : shaderDataTypeConversionLut) {
+                if (e.asDataType == dataType) {
+                    return e.asStr;
+                }
+            }
+            return null;
+        }
+
+        public static boolean hasUniformType(String source, ShaderDesc.ShaderDataType dataType) {
+            String strDataType = shaderTypeToString(dataType);
+            if (strDataType == null) {
+                return false;
+            }
+
+            String[] inputLines = regexLineBreakPattern.split(source);
+
+            for(String line : inputLines) {
+                if(line.contains("uniform") && !line.contains("{")) {
+                    Matcher uniformMatcher = regexUniformKeywordPattern.matcher(line);
+                    if(uniformMatcher.find()) {
+                        String keyword = uniformMatcher.group("keyword");
+                        if (keyword != null)
+                        {
+                            String type = uniformMatcher.group("type");
+                            if (type.equals(strDataType))
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                }
```

Comment:
Same uniform parsing as we do for the ES2 -> ES3 transform. We need to be sure the file has a valid uniform and not just an occurance of the "sampler2DArray" keyword